middleware/capacity.go

```diff
@@ -1,33 +1,92 @@
 package middleware
 
 import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
 	"net/http"
 	"sync/atomic"
 
 	"github.com/julienschmidt/httprouter"
 	"github.com/livepeer/catalyst-api/config"
+	"github.com/livepeer/catalyst-api/handlers"
 	"github.com/livepeer/catalyst-api/metrics"
 	"github.com/livepeer/catalyst-api/pipeline"
 )
 
 type CapacityMiddleware struct {
-	requestsInFlight atomic.Int64
+	vodRequestsInFlight  atomic.Int64
+	clipRequestsInFlight atomic.Int64
 }
 
 func (c *CapacityMiddleware) HasCapacity(vodEngine *pipeline.Coordinator, next httprouter.Handle) httprouter.Handle {
 	return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+		var vodJobCount, clipJobCount int
+
 		// Keep a gauge of HTTP requests in flight
 		metrics.Metrics.HTTPRequestsInFlight.Add(1)
 		defer metrics.Metrics.HTTPRequestsInFlight.Add(-1)
 
-		requestsInFlight := c.requestsInFlight.Add(1)
-		defer c.requestsInFlight.Add(-1)
+		// Get total in-progress VOD jobs
+		allVodJobs := vodEngine.Jobs.GetJobs()
+		vodJobCount = len(allVodJobs)
```

Comment:
Shouldn't `vodJobCount` be the jobs that are NOT clips? Otherwise, IIUC, what could happen on a worst case is that the clip jobs will starve the capacity for VOD jobs. 

e.g.: we have 10 clips running concurrently all the time, with clips finishing but new ones coming in. This means the `vodJobCount` here will always be higher than the limit, and only clip jobs will be accepted.

I guess this could be fixed by calculating `vodJobCount++` an an `else` around line 39 instead, when the given jobs is not a clip. WDYT?

Reply:
Yeah I deleted some logic that accounted for that condition while i was refactoring the PR for review. Fixing now and what you suggested is simpler.